controllers/clusterurlmonitor/clusterurlmonitor_supplement.go

```diff
@@ -0,0 +1,99 @@
+package clusterurlmonitor
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/route-monitor-operator/api/v1alpha1"
+	"github.com/openshift/route-monitor-operator/pkg/blackboxexporter"
+	"github.com/openshift/route-monitor-operator/pkg/const/blackbox"
+	utilfinalizer "github.com/openshift/route-monitor-operator/pkg/util/finalizer"
+	"github.com/openshift/route-monitor-operator/pkg/util/reconcile"
+	"github.com/openshift/route-monitor-operator/pkg/util/templates"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ClusterUrlMonitorSupplement struct {
+	clusterUrlMonitor v1alpha1.ClusterUrlMonitor
+	client            client.Client
+	log               logr.Logger
+	ctx               context.Context
+	blackboxExporter  *blackboxexporter.BlackboxExporter
+}
+
+func NewSupplement(clusterUrlMonitor v1alpha1.ClusterUrlMonitor, client client.Client, log logr.Logger, blackboxExporter *blackboxexporter.BlackboxExporter) *ClusterUrlMonitorSupplement {
+	return &ClusterUrlMonitorSupplement{clusterUrlMonitor, client, log, context.Background(), blackboxExporter}
+}
+
+func (s *ClusterUrlMonitorSupplement) EnsureFinalizer() (reconcile.Result, error) {
+	if !utilfinalizer.Contains(s.clusterUrlMonitor.GetFinalizers(), FinalizerKey) {
+		utilfinalizer.Add(&s.clusterUrlMonitor, FinalizerKey)
+		err := s.client.Update(s.ctx, &s.clusterUrlMonitor)
+		return reconcile.RequeueReconcileWith(err)
+	}
+	return reconcile.ContinueReconcile()
+}
+
+func (s *ClusterUrlMonitorSupplement) EnsureServiceMonitor() (*monitoringv1.ServiceMonitor, error) {
```

Comment:
I feel this function name is cut in the middle "ensureservicemonitor<SOMETHING>" but that may be because I am used to reading it

Reply:
how would you call the function?